### ISSUE TITLE
VoiceOver form shortcuts not working when legend is within some elements

### ISSUE THREAD

**Issue by selfthinker** [alphagov/reported-bugs#33]
**Upstream bug:** https://bugs.webkit.org/show_bug.cgi?id=174936
**Date:** 2017-01-31
**Reported by:** Anika Henke
**Related to:** n/a

---

### Overview

When a legend is within a main (or header, nav or aside) element, it won't be read out if form items are navigated by the VO+Cmd+J shortcut on macOS. When they are navigated via the rotor 'form controls' navigation on iOS, a legend within a footer or article element won't work either. Of new semantic elements I tried, only section seems to be fine in both.

### Steps to reproduce

1. Activate VoiceOver (macOS)
2. Open http://jsbin.com/gozawow in Safari
3. Use VO+Cmd+J to navigate to the next form element

### Actual result

The first item (which is a fieldset) is being recognised as being a group but the legend is not read out at all.

### Expected result

When the virtual cursor reaches the fieldset, the legend should be read out. (Like in the second example on the same page.)

### Why is this a problem?

When form elements are grouped with a fieldset, it's usually because their own labels don't make sense out of context. The context is usually given with the legend. For example a group of two radio buttons saying "Yes" and "No" doesn't make sense without the legend which might say "Do you already have an account?". Or three input fields labelled "Day", "Month" and "Year" won't make sense without a legend saying "What is your date of birth?".
You can still read the legend by arrowing up or down or tabbing to the inputs, but some VO users will be used to the shortcuts and might not explore other options.

The above is the description for macOS. When using iOS, put the rotor on iOS to 'Form Controls' and swipe down. The first item is the first radio button and only its label is being read out (when the legend is within the main element). The first item should instead be read out together with the legend before its label (as with the second example within a div).
